resources/pear.rb

```diff
@@ -18,21 +18,80 @@
 # limitations under the License.
 #
 
-default_action :install
-actions :install, :upgrade, :remove, :purge, :reinstall
-
-state_attrs :channel,
-            :directives,
-            :options,
-            :package_name,
-            :preferred_state,
-            :version,
-            :zend_extensions
-
-attribute :package_name, kind_of: String, name_attribute: true
-attribute :version, default: nil
-attribute :channel, kind_of: String
-attribute :options, kind_of: String
-attribute :directives, kind_of: Hash, default: {}
-attribute :zend_extensions, kind_of: Array, default: []
-attribute :preferred_state, default: 'stable'
+property :package_name, String, name_property: true
+property :version, [String, nil], default: nil
+property :channel, String
+property :options, String
+property :directives, Hash, default: {}
+property :zend_extensions, Array, default: []
+property :preferred_state, String, default: 'stable'
+property :binary, String, default: 'pear'
+
+include PhpCookbook::Helpers
+
+load_current_value do |new_resource|
+  unless current_installed_version(new_resource).nil?
+      version(current_installed_version(new_resource))
+      Chef::Log.debug("Current version is #{version}") if version
+  end
+end
+
+action :install do
+  # If we specified a version, and it's not the current version, move to the specified version
+  install_version = new_resource.version unless new_resource.version.nil? || new_resource.version == current_resource.version
+
+  # If it's not installed at all or an upgrade, install it
+  if install_version || new_resource.version.nil?
+    description = "install package #{new_resource.package_name} #{install_version}"
```

Comment:
just collapse this down and avoid setting the variable

Reply:
Copy pasta, was working on getting the current value fixed :)